rclcpp/include/rclcpp/node_interfaces/node_interfaces.hpp

```diff
@@ -150,7 +150,7 @@ class NodeInterfaces
   /// NodeT::SharedPtr Constructor
   template<typename NodeT>
   NodeInterfaces(std::shared_ptr<NodeT> node)  // NOLINT(runtime/explicit)
-  : NodeInterfaces(node ? *node : throw std::invalid_argument("given node pointer is nullptr"))
+  : NodeInterfaces(*node)
```

Comment:
Would this work instead? I'm not sure how to test it on RHEL

```suggestion
  : NodeInterfaces(node ? std::move(*node) : throw std::invalid_argument("given node pointer is nullptr"))
```

Reply:
From https://stackoverflow.com/questions/68127182/c-ternary-operator-calls-copy-constructor-instead-of-move-constructor/68127366#68127366